src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RemoteTagHelperSearchEngine.cs

```diff
@@ -79,4 +83,22 @@ internal sealed class RemoteTagHelperSearchEngine : ITagHelperSearchEngine
 
         return null;
     }
+
+    private static string GetTypeNameForNavigation(TagHelperDescriptor boundTagHelper)
+    {
+        // View components type name starts with "__Generated" for some reason, so we need to use the original type name metadata instead
+        if (boundTagHelper.Metadata is ViewComponentMetadata
+            {
+                OriginalTypeName:
+                {
+                    IsNull: false,
+                    FullName: { } originalTypeName
+                }
+            })
+        {
+            return originalTypeName;
+        }
+
+        return boundTagHelper.TypeName;
```

Comment:
If you accept my other suggestions, you don't need to be fancy here. It should be enough to just do...

```suggestion
        return ((ViewComponentMetadata)boundTagHelper.Metadata).OriginalTypeName;
```